lib-extra/src/main/java/com/diffplug/spotless/extra/config/SemanticVersion.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.extra.config;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.Objects;
+import java.util.Optional;
+
+/** Semantic version configuration */
+class SemanticVersion implements Serializable, Comparable<SemanticVersion> {
+
+	public static final String FORMAT = "<major>[.minor][.patch]";
+	private static final int MAJOR = 0;
+	private static final int MINOR = 1;
+	private static final int PATCH = 2;
+
+	// Not used, only the serialization output is required to determine whether the object has changed
+	private static final long serialVersionUID = 1L;
+
+	private final int[] version;
+
+	/** Convert string into semantic version */
+	SemanticVersion(final String version) {
+		if (null == version) {
+			throw new UserArgumentException(null, "Version information");
+		}
+		LinkedList<String> versionParts = new LinkedList<String>(
+				Arrays.asList(version.split("\\.", -1)));
```

Comment:
Ditto wrt. `LinkedList`.

Reply:
:-1: 
I use addLast here. Performance does not matter. I just find the code neater. But let me know if you have a more elegant way to fill up missing minor/patch entries.

Reply:
In that case, try `Deque` and `ArrayDeque` instead. :)